src/transformers/models/wav2vec2_bert/processing_wav2vec2_bert.py

```diff
@@ -0,0 +1,142 @@
+                FutureWarning,
+            )
+
+            feature_extractor = SeamlessM4TFeatureExtractor.from_pretrained(pretrained_model_name_or_path, **kwargs)
+            tokenizer = Wav2Vec2CTCTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
+
+            return cls(feature_extractor=feature_extractor, tokenizer=tokenizer)
+
+    def __call__(self, audio=None, text=None, **kwargs):
+        """
+        Main method to prepare for the model one or several sequences(s) and audio(s). This method forwards the `audio`
+        and `kwargs` arguments to SeamlessM4TFeatureExtractor's [`~SeamlessM4TFeatureExtractor.__call__`] if `audio` is not
+        `None` to pre-process the audio. To prepare the target sequences(s), this method forwards the `text` and `kwargs` arguments to
+        PreTrainedTokenizer's [`~PreTrainedTokenizer.__call__`] if `text` is not `None`. Please refer to the doctsring of the above two methods for more information.
+
+        Args:
+            text (`str`, `List[str]`, `List[List[str]]`):
+                The sequence or batch of sequences to be encoded. Each sequence can be a string or a list of strings
+                (pretokenized string). If the sequences are provided as list of strings (pretokenized), you must set
+                `is_split_into_words=True` (to lift the ambiguity with a batch of sequences).
+            audio (`np.ndarray`, `torch.Tensor`, `List[np.ndarray]`, `List[torch.Tensor]`):
+                The audio or batch of audios to be prepared. Each audio can be NumPy array or PyTorch tensor. In case
+                of a NumPy array/PyTorch tensor, each audio should be of shape (C, T), where C is a number of channels,
+                and T the sample length of the audio.
+            kwargs (*optional*):
+                Remaining dictionary of keyword arguments that will be passed to the feature extractor and/or the
+                tokenizer.
+        Returns:
+            [`BatchEncoding`]: A [`BatchEncoding`] with the following fields:
+
+            - **input_features** -- Audio input features to be fed to a model. Returned when `audio` is not `None`.
+            - **attention_mask** -- List of indices specifying which tokens should be attended to by the model (when
+              `return_attention_mask=True` or if *"attention_mask"* is in `self.model_input_names` and if `text` is not
+              `None`).
+            - **input_ids** -- List of token ids to be fed to a model. Returned when `text` is not `None`.
+        """
+
+        sampling_rate = kwargs.pop("sampling_rate", None)
+
+        if audio is None and text is None:
+            raise ValueError("You need to specify either an `audio` or `text` input to process.")
+        elif text is not None and audio is not None:
+            raise ValueError(
+                "`Text` and `audio` are mututally exclusive when passed to `Wav2Vec2BertProcessor`. Specify one or another."
+            )
```

Comment:
Currently, it's mutually exclusive, so not possible. I believe it remove the hassle of understanding which kwargs goes with which class. Note that the current usage seems to pad each modality independently (e.g [here](https://github.com/huggingface/transformers/blob/ff86bc364d20a4e6e3c0d954485eab959c803394/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py#L319-L331)).

Of course, open to removing this exclusivity and make it compatible to both modalities at the same time!

Reply:
The reason I ask is whether we need a processor at all i.e. are there tasks which need both the feature extractor and tokenizer? It's fine if we need one for input and one for output - but if the input/output pairs only ever use one, then we shouldn't use a processor class

Reply:
I've updated the code to go back to Wav2Vec2Processor's original [behaviour](https://github.com/huggingface/transformers/blob/0cdcd7a2b319689d75ae4807cfb7b228aa322f83/src/transformers/models/wav2vec2/processing_wav2vec2.py#L128-L134). 

What bothers me is that the label attention mask is not used in the "forward" and "pad" methods when both modalities are passed. From the [`run_speech_recognition_ctc.py`](https://github.com/huggingface/transformers/blob/0cdcd7a2b319689d75ae4807cfb7b228aa322f83/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py#L333-L334), 

From the [CTC training example](https://github.com/huggingface/transformers/blob/0cdcd7a2b319689d75ae4807cfb7b228aa322f83/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py#L333-L334) and from looking at the `ForCTC` models, I reckon that labels are padded with `-100`. I can maybe adapt this behaviour, but anyways it is something that's supposed to be done in every fine-tuning examples. 

WDYT is the best way to solve this question? I'm personally in favor of staying with the current W2V2's behavior.

cc @sanchit-gandhi for visibility as well

Reply:
> This still has the advantages of being consistent with previous W2V2 model, and having less import, easier from_pretrained/push_to_hub, and less overall complexity as compared to using both classes separately for the end-to-end user IMO.

Being consistent with other w2v2 models makes sense. I'm not sure I see the point about imports and from_pretrained though? If a model only uses a tokenizer, importing, saving and loading it is the same as using a processor. 

Processors are there to bundle together processing objects which are both needed for a single dataset -> inputs -> model -> outputs -> processed_outputs pass - not just for related objects. 

> What bothers me is that the label attention mask is not used in the "forward" and "pad" methods when both modalities are passed.

Could you expand on this further? What exactly is happening and what is the issue caused? 

I wouldn't worry too much about being compatible with training scripts - other than for backwards compatibility. The processor defines behaviour - the training scripts are then demonstrations of how to use them.

Reply:
>  If a model only uses a tokenizer, importing, saving and loading it is the same as using a processor.

The `ForCTC` models use both the tokenizer and feature extractor, so the processor import is easier than importing both the tokenizer and feature extractor here!

> Could you expand on this further? What exactly is happening and what is the issue caused?

I'm taking the example of the original Wav2Vec2 processor, since I've kept same behaviour here for now. 
https://github.com/huggingface/transformers/blob/07ae53e6e77ec6ff4fb25fbacfec4b11cfc82749/src/transformers/models/wav2vec2/processing_wav2vec2.py#L128-L134

As you can see, when both `audio` and `text` are passed, the `text` input_ids are used as a target, but the `text` attention mask is not passed through.
However, if you want to train the CTC model, you have to pad the input ids with a negative value (as shown [here](https://github.com/huggingface/transformers/blob/0cdcd7a2b319689d75ae4807cfb7b228aa322f83/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py#L333-L334) in the examples) because the `text` attention mask is infered in the CTC forward pass (as shown [here](https://github.com/huggingface/transformers/blob/0cdcd7a2b319689d75ae4807cfb7b228aa322f83/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L1988-L1990) in the modeling code).

Note that this CTC behaviour is common across all CTC classes. Also note that the W2V2 processor (and this new processor) are only meant to be used for CTC training.

To sum up, the current behaviour is to that the user himself pad the labels with a negative value with the `text` attention mask. To get it, the users currently do have to pass the audio and text separately. Ideally, the processors should take care of this but the already integrated processors don't.

Reply:
If the current processors only process `text` or `audio`, then I don't think it's breaking anything to enable passing both inputs together and creating the attention mask as necessary - it's just extending what's there. 

The only thing I'd say is that the processor shouldn't be setting values to -100 to be ignored. This is a behaviour we always leave to the user.

Reply:
I've added `label_attention_mask` in [37dd941](https://github.com/huggingface/transformers/pull/28165/commits/37dd941d89a357b720028e8d4be1e0c4c9fd1079), let me know if that works for you. If necessary, I can revert to just adding "labels" when both modalities are passed.

Reply:
Hey @amyeroberts, following our offline discussions, I revert it to just adding `labels` when both modalities are passed.